### ISSUE TITLE
False Positive E1136: PyTorch nn.Parameter.repeat (Value is unsubscriptable)

### ISSUE THREAD

**Issue by enrico-stauss** [pylint-dev/pylint#9311]
### Bug description

```python
import torch

test = torch.nn.Parameter(torch.rand(2, 10, 1, 3))
test = test.repeat(1, 1, 3, 1)[:, :, 0, :]
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:4:7: E1136: Value 'test.repeat(1, 1, 3, 1)' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No warning should be raised as the return of repeat is a `torch.Tensor`. It's possible though, that the issue has to be fixed on the PyTorch side.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.10.4 (main, Sep 11 2022, 20:39:17) [Clang 13.1.6 (clang-1316.0.21.2.5)]
```

### Additional dependencies

```python
torch==2.1.0
```

**Comment by Pierre-Sassoulas** [comment]
Do not reproduce since [pytorch/pytorch 2.8.0](https://github.com/pytorch/pytorch/releases/tag/v2.8.0). But still reproduce with torch 2.1.0 on latest pylint main.
